### DISCUSSION TITLE
[Bug]: Vite is served as insecure

### DISCUSSION THREAD

**Discussion by AlunR** [beyondcode/herd-community#699] [General]
### Platform

macOS

### Operating system version

Mac Sonoma 14.2.1 (23C71)

### System architecture

ARM64 (M1, M2, etc)

### Herd Version

1.6.0 (Build: 24)

### Bug description

When I attempt to run npm run dev the web app fails to load vite resources as the server is reporting as insecure.  If i remove the port :5173 then the server reports as secure

![Screenshot 2024-04-19 at 15 00 51](https://github.com/beyondcode/herd-community/assets/1318497/1eb69e7a-7709-4bd2-8add-816ca0fd42c8)

![Screenshot 2024-04-19 at 15 01 01](https://github.com/beyondcode/herd-community/assets/1318497/e3250e6d-0527-4787-9dab-0ea5b4345580)

### Steps to reproduce

Attempt to use vite resources in an application

**Comment by adrum** [comment]
What version of the Laravel Vite plugin are you using? 1.0 brought automatic Herd TLS detection.

**Comment by bcorcoran** [comment]
I'm having the same issue. Running `herd secure domain.test` works fine but domain.test:5173 still throws an invalid cert error so no resources load (and therefore the app doesn't load) when i'm using `pnpm run dev`.

I tried `herd secure domain.test:5173` and it says it worked but it still has the same issue. I have laravel-vite-plugin @ 1.1.1 and herd 1.13.0

edit: Leaving this here, but the issue was that I never read the changelogs for `laravel-vite-plugin`, so I was using the `server:` option along with the 'old' way of doing things by having a `detectServerConfig(host)` function to specify which certs to use with vite. Removing the `server:` option out of `defineConfig` fixed it since they're handling this for us now.

that said, if you're still using `detectServerConfig` for some reason, you can change the following two vars to use Herd's certs:
```
let keyPath = resolve(homedir(), `Library/Application\ Support/Herd/config/valet/Certificates/${host}.key`)
let certificatePath = resolve(homedir(), `Library/Application\ Support/Herd/config/valet/Certificates/${host}.crt`)
```
